ChangeLog.d/fix_build_for_directory_names_containing_spaces.txt

```diff
@@ -0,0 +1,4 @@
+Bugfix
+   * Fix a bug in the build where directory names containing spaces were
+     causing generate_errors.pl to error out resulting in a build failure.
+     Raised in issue #6879.
```

Comment:
This should probably say "Fixes #6879." and should definitely end with a newline.

Reply:
But it doesn't fully fix #6879 though does it? In @gilles-peskine-arm's [comment](https://github.com/Mbed-TLS/mbedtls/issues/6879#issuecomment-1373712786) he mentioned two bugs. The first being the issue with the script, that this PR addresses and the other being the fact that the Perl script erroring out didn't halt the build.

Should I use this PR to close issue #6879 and then file a separate issue for the fact that errors in generate_errors.pl does not halt the build process?

Reply:
It does fix both issues. The `glob` change makes it work with spaces in the path, and the `or die` change will make it out.

What we saw in the build was a warning on the `<FILE>` input because the `open` didn't work; the `die` means that an error is reported instead, which does halt the build.